ui/dashboards/src/context/DashboardProvider/layout-slice.tsx

```diff
@@ -119,10 +137,27 @@ export function createLayoutSlice(
   return (set, get) => ({
     panelGroups,
     panelGroupIdOrder,
+    previousPanelGroupStates: { panelGroups, panelGroupIdOrder },
 
     // TODO: Reorder init logic so this isn't exposed
     createPanelGroupId,
 
+    savePanelGroups() {
+      set((state) => {
+        state.previousPanelGroupStates = {
+          panelGroups: state.panelGroups,
+          panelGroupIdOrder: state.panelGroupIdOrder,
+        };
+      });
+    },
```

Comment:
Rather than saving old states in the store, what do you think about just assuming that on save, something's going to be creating a new dashboard resource which will cause the store to get re-initialized from scratch? That would simplify all of this quite a bit because you wouldn't need to keep additional state around and "reset" could basically just use the initial state it has from the closure when the store was originally initialized.

Reply:
I'm realizing that in order to do this, all the "initial state" would need to be getting created together (not in separate slices). I had started to centralize that logic in another branch, but I'll think about it some more and maybe we can refactor later to do this.